cachi2/core/package_managers/bundler/parser.py

```diff
@@ -71,7 +71,8 @@ def purl(self) -> str:
     @cached_property
     def remote_location(self) -> str:
         """Return remote location to download this gem from."""
-        return f"{self.source}/gems/{self.name}-{self.version}.gem"
+        # The source URL ends with a slash
+        return f"{self.source}downloads/{self.name}-{self.version}.gem"
```

Comment:
The fact that we'd need a commentary to explain kinda disqualifies the change as is IMO. Unfortunately the precedent we have for platform gems that led you to this change slipped my eye when it was added. The way this should be handled is with `urljoin` from `urllib.parse` which shields us from potentially constructing an invalid URL, e.g. by adding an extra slash (as I TIL would be invalid).